presidio-anonymizer/presidio_anonymizer/entities/text_manipulator.py

```diff
@@ -0,0 +1,47 @@
+"""Handles the original text and creates a new one according to changes requests."""
+from presidio_anonymizer.entities import InvalidParamException
+
+
+# TODO awful name! - PR people any ideas?
+class TextManipulator:
+    """Creates new text according to users request."""
+
+    def __init__(self,
+                 original_text: str):
+        self.original_text = original_text
+        self.output_text = original_text
+        self.text_len = len(original_text)
+        self.last_replacement_point = self.text_len
+
+    def __validate_position_over_text(self, start: int, end: int):
+        if start > self.text_len or end > self.text_len:
+            raise InvalidParamException(
+                f"Invalid text position start with {start} and end with {end}, "
+                f"original text length is only {self.text_len}."
+            )
+
+    def validate_and_get_text_in_position(self, start: int, end: int):
+        """
+        Validate and get part of the text inside the original text.
+
+        :param start: start position of inner text
+        :param end: end position of inner text
+        :return: str - part of the original text
+        """
+        self.__validate_position_over_text(start, end)
+        return self.output_text[start: end]
+
+    def replace_text(self, anonymized_text: str, start: int, end: int):
```

Comment:
Lets talk about it tomorrow please. I want to fully understand this.

Reply:
I want to make sure we keep the anonymizer stateless. Make sure to add me in this discussion and before any decisions are made

Reply:
While we iterate over the text, we need to keep state somewhere - we have a text that we keep modifying during the iteration.
My comment here, looking ahead for when we will add a list of replacement indices, in addition to the final text that keep changing, is have this state inside the 'ResultBuilder'. Hope that's clearer.

Reply:
Nothing changed here in terms of the state or functionality.
The only change was taking the code and moving it to another class so it will be more manageable.